cluster/build.gradle

```diff
@@ -133,6 +133,30 @@ dependencies {
     exclude(group: 'org.apache.directory.server', module: 'apacheds-kerberos-codec')
   }
 
+  compile(group: 'org.apache.hadoop', name: 'hadoop-azure', version: hadoopVersion) {
+    // exclude some dependencies like in hadoop-aws to avoid conflicts with other deps
+    exclude(group: 'asm', module: 'asm')
```

Comment:
Curious to know how did you arrive at these exclusions? Is this the complete list of exclusions or is it something we'll have to update continuously as and when we see any conflicts during test runs, etc.

Reply:
Initially, I just added Guava since we explicitly fetch it elsewhere. However, I saw a problem in statistics collection in the web app, since a library used by Jetty had a different version (indicating that we need to exclude additional dependencies). I later just excluded everything that hadoop-aws excludes since it would be safe to do so.

The list all transitive dependencies of the hadoop-azure library can be obtained following this
https://solidsoft.wordpress.com/2014/11/19/gradle-tricks-tracking-down-not-expected-transitive-dependencies/  

This task prints the dependency tree for snappy-cluster 
`./gradlew :snappy-cluster_2.11:dependencies
`
Snippet from the output for hadoop-aws and hadoop-azure
....
+--- org.apache.hadoop:hadoop-aws:2.7.7
|    +--- org.apache.hadoop:hadoop-common:2.7.7 (*)
|    +--- com.fasterxml.jackson.core:jackson-databind:2.2.3 -> 2.6.7.2 (*)
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.2.3 -> 2.6.7
|    \--- com.amazonaws:aws-java-sdk:1.7.4
|         +--- org.apache.httpcomponents:httpclient:4.2 -> 4.5.6 (*)
|         +--- commons-codec:commons-codec:1.3 -> 1.11
|         +--- com.fasterxml.jackson.core:jackson-core:2.1.1 -> 2.6.7
|         +--- com.fasterxml.jackson.core:jackson-databind:2.1.1 -> 2.6.7.2 (*)
|         +--- com.fasterxml.jackson.core:jackson-annotations:2.1.1 -> 2.6.7
|         \--- joda-time:joda-time:[2.2,) -> 2.10.1
\--- org.apache.hadoop:hadoop-azure:2.7.7
     +--- org.apache.hadoop:hadoop-common:2.7.7 (*)
     +--- com.fasterxml.jackson.core:jackson-core:2.2.3 -> 2.6.7
     +--- org.apache.httpcomponents:httpclient:4.2.5 -> 4.5.6 (*)
     +--- com.microsoft.azure:azure-storage:2.0.0
     |    +--- com.fasterxml.jackson.core:jackson-core:2.2.3 -> 2.6.7
     |    +--- org.slf4j:slf4j-api:1.7.5 -> 1.7.25
     |    \--- org.apache.commons:commons-lang3:3.3.2 -> 3.8.1
     \--- com.google.guava:guava:11.0.2 -> 14.0.1
....